src/lib/all.ts

```diff
@@ -1,21 +1,39 @@
-import { assign, times, flatten } from "lodash"
+import { times, flatten } from "lodash"
 
-export const allViaLoader = (loader, loaderOptions, apiOptions: any = {}) => {
-  const countOptions = assign({}, apiOptions, {
+// TODO: Type this and refactor, because I think things just kinda work by luck,
+//       because a static and a dynamic path loader take different number of
+//       arguments.
+
+export const MAX_GRAPHQL_INT = 2147483647
+
+export const allViaLoader = (
+  loader,
+  options: {
+    // For dynamic path loaders
+    path?: string | { [key: string]: any }
+    params?: { [key: string]: any }
+    api?: { [key: string]: any }
+  } = {}
+) => {
+  const params = options.params ? { size: 25, ...options.params } : { size: 25 }
+  const invokeLoader = invocationParams =>
+    options.path
+      ? loader(options.path, invocationParams, options.api)
+      : loader(invocationParams, options.api)
+  const countParams = {
+    ...params,
     page: 1,
-    size: 0,
+    size: 1, // FIXME: The /shows endpoint does not return a count with size=0
```

Comment:
We need to fix this in Gravity, I presume.

Reply:
so this query is just trying to get the counts?

Reply:
Yup